Neos.Flow/Classes/Http/Request.php

```diff
@@ -590,27 +585,48 @@ function (&$value) {
     }
 
     /**
-     * Returns and array of all possibles "field paths" for the given array.
+     * Returns and array of all possible "field paths" for the given array.
      *
      * @param array $structure The array to walk through
      * @param string $firstLevelFieldName
      * @return array An array of paths (as strings) in the format "key1/key2/key3" ...
      */
-    protected function calculateFieldPaths(array $structure, $firstLevelFieldName = null)
+    protected function calculateFieldPaths(array $structure, string $firstLevelFieldName = null)
+    {
+        $fieldPaths = $this->calculateFieldPathsAsArray($structure, $firstLevelFieldName);
+        array_walk($fieldPaths, function (&$fieldPath) {
+            $fieldPath = implode('/', $fieldPath);
+        });
+        return $fieldPaths;
+    }
+
+    /**
+     * Returns and array of all possible "field paths" for the given array.
+     *
+     * @param array $structure The array to walk through
+     * @param string $firstLevelFieldName
+     * @return array An array of paths (as arrays) in the format ["key1", "key2", "key3"] ...
+     */
+    protected function calculateFieldPathsAsArray(array $structure, string $firstLevelFieldName = null): array
     {
         $fieldPaths = [];
         if (is_array($structure)) {
             foreach ($structure as $key => $subStructure) {
-                $fieldPath = ($firstLevelFieldName !== null ? $firstLevelFieldName . '/' : '') . $key;
+                $fieldPath = [];
```

Comment:
nitpick: you just copied the old code, but we could safe one level of indention by removing the `is_array($structure)`. The type is already verified via the type hint